tidy3d/plugins/design/method.py

```diff
@@ -175,28 +110,466 @@ class MethodGrid(MethodIndependent):
+                f"Best Parameters: {' '.join([f'{param}: {val}' for param, val in opt.max['params'].items()])}\n"
+            )
+
+        # Output fn_args from the BO.opt object - getting results in situ as opt changes type to float
+        fn_args = []
+        for output in opt.res:
+            fn_args.append(output["params"])
+
+        return fn_args, result, total_aux_out, opt
+
+
+class MethodGenAlg(MethodOptimize, ABC):
+    """A standard method for performing genetic algorithm search"""
+
+    # Args for the user
+    solutions_per_pop: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of solutions per population.",
+        description="The number of solutions to be generated for each population.",
+    )
+
+    n_generations: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of generations.",
+        description="The maximum number of generations to run the genetic algorithm.",
+    )
+
+    n_parents_mating: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of parents mating.",
+        description="The number of solutions to be selected as parents for the next generation.",
+    )
+
+    stop_criteria: Optional[pd.constr(regex=r"\b(?:reach|saturate)_\d+\b")] = pd.Field(
+        default=None,
+        title="Early stopping criteria.",
+        description="Define the early stopping criteria. Supported words are 'reach_X' or 'saturate_X' where X is a number. See PyGAD docs for more details.",
+    )
+
+    parent_selection_type: Optional[
+        Literal["sss", "rws", "sus", "rank", "random", "tournament"]
+    ] = pd.Field(
+        default="sss",
+        title="Parent selection type.",
+        description="The style of parent selector. See the PyGAD docs for more details.",
+    )
+
+    crossover_type: Optional[Literal["single_point", "two_points", "uniform", "scattered"]] = (
+        pd.Field(
+            default="single_point",
+            title="Crossover type.",
+            description="The style of crossover operation. See the PyGAD docs for more details.",
+        )
+    )
+
+    crossover_prob: Optional[pd.confloat(ge=0, le=1)] = pd.Field(
+        default=0.8,
+        title="Crossover probability.",
+        description="The probability of performing a crossover between two parents.",
+    )
+
+    mutation_type: Optional[Literal["random", "swap", "inversion", "scramble", "adaptive"]] = (
+        pd.Field(
+            default="random",
+            title="Crossover type.",
+            description="The style of gene mutation.",
+        )
+    )
+
+    mutation_prob: Optional[pd.confloat(ge=0, le=1)] = pd.Field(
+        default=0.2,
+        title="Crossover probability.",
+        description="The probability of mutating a gene.",
+    )
+
+    # TODO: See if anyone is interested in having the full suite of PyGAD options - there's a lot!
+
+    def run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable, console) -> Tuple[Any]:
```

Comment:
docstring

Reply:
one thing to consider is allowing  `**kwargs` to be passed here from `design.run()`. then the user could theoretically have these passed directly to `pygad.GA`. (similar for BO and other methods).